### PR TITLE
Add zsh framework plugin support

### PR DESCRIPTION
**Pull request by unixorn**
* Added a ZSH completion function for the **desk** command
* Added a plugin.zsh file. This will allow frameworks like [Antigen](https://github.com/zsh-users/antigen), [oh-my-zsh](http://ohmyz.sh/), and [zgen](https://github.com/tarjoilija/zgen) to automatically clone the repository and add it to the user's `$PATH`.
* Updated installation instructions to include sections for Antigen, oh-my-zsh and zgen
* Cleaned up some trailing spaces on lines in **README.md**.

### CODE REVIEW

File: completions/_desk [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#compdef desk
+_desk() {
+  local -a commands
+  commands=(
+    'help:Print a help message.'
+    'init:Initialize your desk configuration.'
+    'list:List available desks'
+    'ls:List available desks'
+    'edit:Edit or create a desk, defaults to current desk'
+    'go:activate a desk'
```
Comment (jamesob) [inline completions/_desk]: I think we're missing the `.` alias for `go`

**Comment by jamesob** [comment]
@unixorn rad! Thanks for the PR! Looks good -- just reading up on `.zsh` plugins so I can merge this credibly ;)

**Comment by jamesob** [comment]
I take it you've tested this with antigen/oh-my-zsh/zgen?

**Comment by unixorn** [comment]
I've tested on zgen since that's the framework I use. The plugin framework is standard, so If it works with one framework, it should work with all of them.

**Comment by jamesob** [comment]
:+1: thanks much, @unixorn.

**Pull request merged**